packages/@aws-cdk/aws-apigateway/lib/api-definition.ts

```diff
@@ -82,6 +83,15 @@ export abstract class ApiDefinition {
    * assume it's initialized. You may just use it as a construct scope.
    */
   public abstract bind(scope: Construct): ApiDefinitionConfig;
+
+  /**
+   * Called after the CFN RestApi resource has been created to allow the Api
+   * Definition to bind to it. Specifically it's required to allow assets to add
+   * metadata for tooling like SAM CLI to be able to find their origins.
+   */
+  public bindToResource(_resource: CfnResource) {
```

Comment:
Can we make this: `bindAfterCreate(scope: Construct, restApi: RestApi)` ?

Call it like `apiDefinition.bindAfterCreate(this, this)`, and you can access the CfnResource inside like:

```ts
const child = Node.of(restApi).defaultChild as CfnRestApi;
child.addMetadata(...);
```
```

Reply:
sure